### DISCUSSION TITLE
Docs that cover Craft setup & plugin testing

### DISCUSSION THREAD

**Discussion by sjelfull** [markhuot/craft-pest#103] [General]
Hi Mark, love the look of this! I'd love to get started testing with Pest, but the docs is missing some fundamental things that makes me confused:
- I'd love to see a section that covers setting up a Craft instance, as well as creating a database dump to use as foundation.
- Also would like to see more materials around plugin testing
- Third, if possible, I hope to use ☝️ to start testing my plugins, and ideally run tests in CI on pull requests. Anything like an example project or docs that cover that would be very welcome!

**Comment by markhuot** [comment]
Those are all really good suggestions. I don't have time to integrate each of them right now but hopefully these links get you on your way,

1. Installing Craft is covered pretty well on their docs, https://craftcms.com/docs/4.x/installation.html#quick-start
2. You shouldn't need a database export. By default `craft-pest` runs against your local database with transactions to keep test data from filtering in to your real content.
3. Plugin testing is on my roadmap, although this plugin itself is tested with Pest. You can see a lot of examples in the `tests` folder of this project. For example, https://github.com/markhuot/craft-pest/blob/master/tests/HttpCpTest.php
4. I also run tests in CI right here in this plugin. See, https://github.com/markhuot/craft-pest/blob/7334804cf50921575dfa497205cce7bdba36f3a8/.github/workflows/php.yml#L98-L99
